### ISSUE TITLE
Bug: [no-floating-promises] support TsAsExpression

### ISSUE THREAD

**Issue by arka1002** [typescript-eslint/typescript-eslint#8692]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.2&fileType=.tsx&code=C4TwDgpgBAYg9nKBeKAFATnAtgSwM4QA8AdgK5YBGE6AfFAGRQDeUAFhCAPwBcUew6HMQDmUAL4BuALAAoADYRgUCgEMAXmuRQVeEMQDGUABQBKZHQBM0mao2nteWAglA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiRwYAL4h1QA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Foo = Promise<number> & { hey?: string };
let bazz = async () => 2;
bazz() as Foo;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-floating-promises": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that the 3rd line should report the absence of handling the rejection of the promise.

### Actual Result

There was no error on 3rd line

**Comment by arka1002** [comment]
Oh! its very similar to https://github.com/typescript-eslint/typescript-eslint/issues/8654
